### PR TITLE
Use rpcmem instead of custom ION allocator.

### PR DESCRIPTION
**Pull request by dsharletg**
This change removes the ION allocator from the halide runtime, and uses the rpcmem lib from the Qualcomm SDK instead. This seems to work better for large, complicated pipelines that allocate several large buffers.

### CODE REVIEW

File: src/runtime/hexagon_remote/Makefile [hunk 1]
```diff
@@ -41,6 +41,17 @@ COMMON_FLAGS = -I ${HEXAGON_SDK_INCLUDES}/stddef
 COMMON_CCFLAGS = ${COMMON_FLAGS} -O3 -I ${HEXAGON_SDK_LIBS}/common/remote/ship/android_Release
 HEXAGON_QAICFLAGS = ${COMMON_FLAGS}
 
+# rpcmem doesn't have a precompiled library in the SDK, so we just
```
Comment (dsharletg) [inline src/runtime/hexagon_remote/Makefile]: @ronlieb @pranavb-ca @dpalermo Did I just miss something here?

**Comment by dsharletg** [comment]
Actually, this seems to have Hexagon code 25% slower. Hold on.

**Comment by dsharletg** [comment]
OK, the performance issue is fixed. Please take a look at this, it's somewhat urgent that I get this committed. The vast majority of this is removing code, the only interesting change is in the Makefile.

**Comment by abadams** [comment]
Looks like this strictly deletes code from libHalide itself. Feel free to merge once travis goes green.

File: src/runtime/hexagon_remote/Makefile (hunk 1 above)
Comment (ronlieb) [inline src/runtime/hexagon_remote/Makefile]: it seems like you figured it out. We should just push it thru and then do additional testing here on MTP using our toyCamera and HelloHexagon  as well as our usual tests.

**Pull request merged**